pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -72,6 +72,17 @@ public void completeSegmentOperations(String tableNameWithType, SegmentMetadata
       @Nullable String sourceDownloadURIStr, String segmentDownloadURIStr, @Nullable String crypterName,
       long segmentSizeInBytes, boolean enableParallelPushProtection, boolean allowRefresh, HttpHeaders headers)
       throws Exception {
+    completeSegmentOperations(tableNameWithType, segmentMetadata, uploadType, finalSegmentLocationURI, segmentFile,
```

Comment:
Let's not coupling the logic of reingestion into current regular segment complete handling. Currently the logic is very hard to read, and we are coupling reset with reingestion, which is not correct.
We can add a new method `completeReingestedSegmentOperations()`. Reingested segment holds different assumptions then regular uploaded segment.

Reply:
done